src/com/goide/inspections/unresolved/GoUnusedParameterInspection.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections.unresolved;
+
+import com.goide.inspections.GoInspectionBase;
+import com.goide.psi.*;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.search.searches.ReferencesSearch;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.Query;
+import org.jetbrains.annotations.NotNull;
+
+public class GoUnusedParameterInspection extends GoInspectionBase {
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder,
+                                     @SuppressWarnings({"UnusedParameters", "For future"}) @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitParameters(@NotNull GoParameters o) {
+        super.visitParameters(o);
+        if (o.getParameterDeclarationList().size() == 0) return;
+        if (PsiTreeUtil.getParentOfType(o, GoInterfaceType.class, GoTypeDeclaration.class, GoConversionExpr.class, GoTypeCaseClause.class) != null) return;
+        GoFunctionType funcType = PsiTreeUtil.getParentOfType(o, GoFunctionType.class);
```

Comment:
I think it's better to override visitFunctionDeclaration() and retrieve signature/parameters/result/etc. for GoFunctionDeclaration instead of visiting parameters and going though PSI tree looking for needed elements